lightning-invoice/src/lib.rs

```diff
@@ -1235,6 +1240,21 @@ impl Invoice {
 		}
 	}
 
+	/// Returns the Duration remaining until the invoice expires.
+	#[cfg(feature = "std")]
+	pub fn expiration_remaining(&self) -> Option<Duration> {
```

Comment:
I think we could make this a non-`Option` and return `Duration::(0)` on error. Similar below

Reply:
Should we not use `checked_sub` then and allow the panic for overflows?